src/util/Conversions.h

```diff
@@ -568,10 +571,37 @@ inline string removeLeadingZeros(const string& orig) {
 }
 
 // _____________________________________________________________________________
-bool isXsdValue(const string val) {
+bool isXsdValue(const string& val) {
   // starting the search for "^^ at position 1 makes sure it's not in the
   // quotes as we already checked that the first char is the first quote
-  return val.size() > 0 && val[0] == '\"' &&
-         val.find("\"^^", 1) != string::npos;
+  return !val.empty() && val[0] == '\"' && val.find("\"^^", 1) != string::npos;
+}
+
+// _____________________________________________________________________________
+bool isNumeric(const string& val) {
+  if (val.empty()) {
+    return false;
+  }
+  size_t start = (val[0] == '-') ? 1 : 0;
+  size_t posNonDigit = val.find_first_not_of("0123456789", start);
+  if (posNonDigit == string::npos) {
+    return true;
+  }
+  if (val[posNonDigit] == '.') {
+    return posNonDigit + 1 < val.size() &&
+           val.find_first_not_of("0123456789", posNonDigit + 1) == string::npos;
+  }
+  return false;
```

Comment:
This currently lacks the support of Floating point numbers of the format `-3.25E8`.  I am not sure though how relevant they are for our use cases.

Reply:
I'd propose deferring this to actually fixing the parser using the knowledge gained from your Turtle parsing.

Reply:
Yes, that is probably true. Maybe we will consider to update this whole Value Literal representation business anyway.